vpr/src/device/rr_graph_builder.h

```diff
@@ -218,6 +220,31 @@ class RRGraphBuilder {
     inline void init_fan_in() {
         node_storage_.init_fan_in();
     }
+    /** @brief Return detailed routing segment information with a given id* @note The routing segments here may not be exactly same as those defined in architecture file. They have been
+     * adapted to fit the context of routing resource graphs.
+     */
+
+    inline const t_segment_inf& rr_segments(RRSegmentId seg_id) const {
```

Comment:
``RRGraphBuilder`` is designed to modify routing resource graph data structure.
Therefore the API required here is to add new rr_segment, such as

```
inline void add_rr_segment(t_segment_inf);
```

We will use this API to replace all the lines in current rr_graph.cpp when adding rr_segments

Same for the ``rr_switch``.